examples/passport-login/src/authentication-strategies/session.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-passport-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {AuthenticationStrategy, asAuthStrategy} from '@loopback/authentication';
+import {RedirectRoute, RequestWithSession, HttpErrors} from '@loopback/rest';
+import {UserProfile, securityId} from '@loopback/security';
+import {User} from '../models';
+import {bind} from '@loopback/context';
+import {repository} from '@loopback/repository';
+import {UserRepository} from '../repositories';
+
+@bind(asAuthStrategy)
+export class SessionStrategy implements AuthenticationStrategy {
+  name = 'session';
+
+  constructor(
+    @repository(UserRepository)
+    public userRepository: UserRepository,
+  ) {}
+
+  /**
+   * authenticate a request
+   * @param request
+   */
+  async authenticate(
+    request: RequestWithSession,
+  ): Promise<UserProfile | RedirectRoute | undefined> {
+    if (!request.user) {
```

Comment:
hmm why the session strategy doesn't retrieve the user from request's session but from request directly?

Reply:
agreed